Add slashtag support to SnuOwnd

JordanMilne opened the pull request:
:eyeglasses: @ajacksified

This is the JS version of https://github.com/reddit/snudown/pull/72. It's more or less a direct translation of the C code to JS, so don't feel like you need to dig too far into the logic, just want to make sure I haven't made a major JS faux pas :smile: .

JordanMilne committed:
Fix multis and reddit.com in sr autolinking

JordanMilne committed:
Fix entity tests

JordanMilne committed:
Add slashtag support

ajacksified commented on test.js:
```diff
@@ -156,9 +156,94 @@ cases = {
+        '<p><a href="/r/all-minus-something">/r/all-minus-something</a></p>\n',
+
+    '/r/notall-minus':
+        '<p><a href="/r/notall">/r/notall</a>-minus</p>\n',
```
Why does the above test case include the whole string, but this one does not?

JordanMilne replied:
`/r/all-` is a special case for the feature that allows you to filter certain subreddits out of `/r/all`. `-` in an old-style multireddit doesn't mean anything outside of that case (`/r/notall-minus` would mean get a listing from `/r/notall` while filtering out results from `/r/minus`,) so we don't autolink it. [Snudown has the same behaviour](https://github.com/reddit/snudown/blob/master/test_snudown.py#L163).

ajacksified replied:
Aha, got it. :+1:

ajacksified commented:
Looks good, just one question. :fish:

JordanMilne committed:
Don't rewind over previous inlines when autolinking

This is the sister commit of https://github.com/reddit/snudown/commit/c4266fd890b4ed0b6d23ddd281e3bd63e433110c
